File: tests/fixtures/golden/dart/sample.dart
import 'dart:async';

abstract class Identifiable {
    String get id;
}

enum Status {
    active,
    inactive
}

class BaseEntity {
    final String name;
    BaseEntity(this.name);
}

class User extends BaseEntity implements Identifiable {
    @override
    final String id;
    Status status = Status.active;

    User(this.id, String name) : super(name);

    String process() {
        return helperCall(name);
    }

    String helperCall(String input) {
        return 'processed_$input';
    }
}

File: tests/fixtures/golden/swift/sample.swift
import Foundation

protocol Identifiable {
    var id: String { get }
}

enum Status {
    case active
    case inactive
}

class BaseEntity {
    var name: String
    init(name: String) {
        self.name = name
    }
}

class User: BaseEntity, Identifiable {
    var id: String
    var status: Status = .active

    init(id: String, name: String) {
        self.id = id
        super.init(name: name)
    }

    func process() -> String {
        return helperCall(input: name)
    }

    private func helperCall(input: String) -> String {
        return "processed_\(input)"
    }
}
